### ISSUE TITLE
preg_match_all $flags default value is unclear

### ISSUE THREAD

**Issue by biziclop** [php/doc-en#1738]
From manual page: https://php.net/function.preg-match-all

Hi, just a small thing, the Description section says that `int $flags = 0,`, but the Parameters section describing `flags` says at the end that: 
>  If no order flag is given, `PREG_PATTERN_ORDER` is assumed. 

The numeric value of `PREG_PATTERN_ORDER` is 1, the source says `$flags = 0` at  
https://github.com/php/php-src/blob/master/ext/pcre/php_pcre.stub.php#L105  
> `function preg_match_all(string $pattern, string $subject, &$matches = null, int $flags = 0, int $offset = 0): int|false {}`

Maybe they could be made consistent, even lying a bit in the docs by saying `$flags = PREG_PATTERN_ORDER`?

**Comment by kocsismate** [comment]
You are right, we can definitely say that the default is `PREG_PATTERN_ORDER`. As the signatures in the documentation are generated based on the stubs, I'll fix this there first so that the docs can occasionally be syncronized with it (sometime soon).

**Comment by cmb69** [comment]
> You are right, we can definitely say that the default is `PREG_PATTERN_ORDER`.

While that would be nice, [it would unfortunately not be true](https://github.com/php/php-src/blob/614e9ec8042681b836703b696d30afae8965e575/ext/pcre/php_pcre.c#L1139-L1140). See the `use_flags` parameter of `php_pcre_match_impl()`.

**Comment by kocsismate** [comment]
> While that would be nice, [it would unfortunately not be true](https://github.com/php/php-src/blob/614e9ec8042681b836703b696d30afae8965e575/ext/pcre/php_pcre.c#L1139-L1140). See the use_flags parameter of php_pcre_match_impl().

Ohh, right. I missed this parameter :S
